js/api/Surveys.js

```diff
@@ -148,6 +169,21 @@ function clearSurveyCache(exclusions) {
     }
 
     realm.write(() => {
+      const expiredSurveysLength = expiredSurveys.length;
+      for (var i = 0; i < expiredSurveysLength; i++) {
+        let forms = realm.objects('Form').filtered(`surveyId= "${expiredSurveys[i].surveyId}"`)
+        let timeTriggers = realm.objects('TimeTrigger').filtered(`surveyId= "${expiredSurveys[i].surveyId}"`)
+        let geofenceTriggers = realm.objects('GeofenceTrigger').filtered(`surveyId= "${expiredSurveys[i].surveyId}"`)
+
+        let formsLength = forms.length;
+        for (var j = 0; j < formsLength; j++) {
+          let questions = realm.objects('Question').filtered(`formId= "${expiredSurveys[i].surveyId}"`)
```

Comment:
It does change quite fast. For a very long while using for loops was significantly faster than forEach and .map()

It always depends on the JS engine/node ver. etc.

I'll keep you posted. The performance issues are supposed to be related to debugging only anyway, it's not something that should affect production.